src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -27,37 +29,16 @@ internal DateTime StartTimeCore
         {
             get
             {
-                // Get the RUsage data and convert the process start time.
-                // To calculcate the process start time in OSX we call proc_pid_rusage() and use ri_proc_start_abstime.
-                //
-                // ri_proc_start_abstime is absolute time which is the time since some reference point. The reference point
-                // usually is the system boot time but this is not necessary true in all versions of OSX. For example in Sierra 
-                // Mac OS 10.12 the reference point is not really the boot time. To be always accurate in our calculations
-                // we don’t assume the reference point to be the boot time and instead we calculate it by calling GetTimebaseInfo
-                // which is a wrapper for native mach_absolute_time(). That method returns the current time referenced to the needed start point
-                // Then we can subtract the returned time from DateTime.UtcNow and we’ll get the exact reference point.
-                //
-                // The absolute time is measured by the bus cycle of the processor which is measured in nanoseconds multiplied 
-                // by some factor “numer / denom”. In most cases the factor is (1 / 1) but we have to get the factor and use it just 
-                // in case we run on a machine has different factor. To get the factor we call GetTimebaseInfo the wrapper for the 
-                // mach_timebase_info() which give us the factor. Then we multiply the factor by the absolute time and the divide
-                // the result by 10^9 to convert it from nanoseconds to seconds.
-
                 EnsureState(State.HaveNonExitedId);
+                Interop.libproc.proc_taskallinfo? info = Interop.libproc.GetProcessInfoById(Id);
 
-                uint numer, denom;
-                Interop.Sys.GetTimebaseInfo(out numer, out denom);
-                Interop.libproc.rusage_info_v3 info = Interop.libproc.proc_pid_rusage(_processId);
-                ulong absoluteTime;
+                if (info == null)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
 
-                if (!Interop.Sys.GetAbsoluteTime(out absoluteTime))
-                {
-                    throw new Win32Exception(SR.RUsageFailure);
-                }
+                DateTime startTime = DateTime.UnixEpoch + TimeSpan.FromSeconds(info.Value.pbsd.pbi_start_tvsec + info.Value.pbsd.pbi_start_tvusec / MicrosecondsToSecondsFactor);
```

Comment:
Should the division include a cast to double, e.g. `pbsd.pbi_start_tvusec / (double)MicrosecondsToSecondsFactor`?

Reply:
Related, have you done any manual testing of this to validate the results?  It looks like our automated testing allows for a lot of wiggleroom on accuracy, around 6 seconds worth.

Reply:
For point (1) the factor is already a double. For point (2), I've reached out in the original issue for manual testing beyond CI. 

I'll remove the wiggle room and see what shakes out.

Reply:
> the factor is already a double

So it is. It's a bit strange that the two factors next to each other have different types.

Reply:
Disabling the wiggle room apparently fails on multiple platforms (other than OS X).